### DISCUSSION TITLE
stop image resizing by the default predictor during testing/ inference

### DISCUSSION THREAD

**Discussion by aymanaboghonim** [facebookresearch/detectron2#4588] [Show and tell]
Hello, I want to know how to stop the image resizing process that is happened automatically by the default predictor on the input image during run inference. 
I want to stop this preprocess because it may result in information loss (missing some objects) when using larger images than the model was trained on.
I tried to overwrite the config but it failed `cfg.INPUT.MIN_SIZE_TEST = 0
`

I also edited the cnfig yaml file to set min size of input = 0 but no effect also.

**Comment by kretes** [comment]
Which piece of code do you refer to as when you say 'default predictor'?

Usually this part is done by `ResizeShortestEdge` and if you don't want it - just remove it.

However - models do have the expectations on the size of the input and keep the input cropped/padded to the expected size.

I am not sure about how other models handle it, but in rcnn there is an always-called `preprocess` function on input_images, defined https://github.com/facebookresearch/detectron2/blob/7c2c8fb168a2093ce06a531c1208fba48d2984ec/detectron2/modeling/meta_arch/rcnn.py#L223 , which normalizes pixel values but as well transforms the size of the input as expected by the backbone. It uses padding to make sure the input size is divisable by the appropriate number and the expected square size.

In case you pass an input that is larger - this will not pad but crop. 

See:

```
from detectron2.structures import ImageList
import torch

images = [torch.rand((3, 200, 180))]

img_tr = ImageList.from_tensors(
            images,
            size_divisibility=32,
            padding_constraints={"size_divisibility": 32, "square_size": 128},
        )

img_tr[0].shape

(images[0][:,:128,:128] == img_tr[0]).all()
# tensor(True)
```

However I feel this is an unwanted behaviour, and I would rather get an exception from it as this is silently cropping my input image.

**Reply by aymanaboghonim** [reply]
so how to force the model to use the input image as it is if it is divisible by 32??
My main concern is to run inference on large image say 8 k * 8 k image without any resizing to prevent any info loss .
